src/shared_ptr.hpp

```diff
@@ -0,0 +1,12 @@
+#ifndef SHARED_PTR_HPP
```

Comment:
please use #pragma once, this is  supported by all compilers

Reply:
What about that?
https://ru.stackoverflow.com/questions/630573/ifndef-define-vs-pragma-once

Reply:
That article has no valid points in it